3-Register/src/components/Button.tsx

```diff
@@ -0,0 +1,24 @@
+interface ButtonProps {
+  className?: string;
+  children: JSX.Element;
+  active: boolean;
+  onClick: () => void;
+}
+
+export const Button = ({
+  className,
+  children,
+  active,
+  onClick,
+}: ButtonProps) => {
+  return (
+    <button
+      className={className}
+      type="button"
+      disabled={!active}
+      onClick={onClick}
+    >
+      {children}
+    </button>
+  );
+};
```

Comment:
No need to re-declare al these props which already exist in a `button` element, you can just do this which also allows you to pass any other prop you might need in the future (also, use `disabled` instead of switching it to `active`):

```suggestion
export const Button = (props: ComponentPropsWithoutRef<"button">) => {
  return (
    <button
      type="button"
      {...props}
    >
      {children}
    </button>
  );
};
```